output/cloud/expv2/mapping.go

```diff
@@ -10,21 +11,41 @@ import (
 )
 
 // TODO: unit test
-func mapTimeSeriesLabelsProto(tags *metrics.TagSet) []*pbcloud.Label {
+func mapTimeSeriesLabelsProto(tags *metrics.TagSet) ([]*pbcloud.Label, []string) {
 	labels := make([]*pbcloud.Label, 0, ((*atlas.Node)(tags)).Len())
+	var discardedLabels []string
 
-	// TODO: move it as a shared func
+	// TODO: move this as a shared func
 	// https://github.com/grafana/k6/issues/2764
 	n := (*atlas.Node)(tags)
 	if n.Len() < 1 {
-		return labels
+		return labels, discardedLabels
 	}
 	for !n.IsRoot() {
 		prev, key, value := n.Data()
-		labels = append(labels, &pbcloud.Label{Name: key, Value: value})
+
 		n = prev
+		if !isReservedLabelName(key) {
+			labels = append(labels, &pbcloud.Label{Name: key, Value: value})
+			continue
+		}
+
+		if discardedLabels == nil {
+			discardedLabels = make([]string, 0, 1)
+		}
+
+		discardedLabels = append(discardedLabels, key)
 	}
-	return labels
+	return labels, discardedLabels
+}
+
+func isReservedLabelName(name string) bool {
+	// this is a reserved label prefix for the prometheus
+	if strings.HasPrefix(name, "__") {
+		return true
+	}
+
+	return name == "test_run_id"
```

Comment:
Feel free to ignore, but I'd suggest putting the static string in a constant.

Reply:
I considered the constant but decided to keep the context tinier (limited by a method). If we use it somewhere else in the application logic, we should move it to the const.